src/astro/sql/operators/agnostic_stats_check.py

```diff
@@ -0,0 +1,392 @@
+from distutils import log as logger
+from os import name, stat
+from typing import Dict, List, Set
+
+from airflow.hooks.base import BaseHook
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from psycopg2 import sql
+from sqlalchemy.sql.expression import table
+from sqlalchemy.sql.functions import Function
+from sqlalchemy.sql.schema import Table
+
+from astro.sql.operators.sql_decorator import SqlDecoratoratedOperator
+from astro.sql.table import Table
+from astro.utils.snowflake_merge_func import (
+    is_valid_snow_identifier,
+    is_valid_snow_identifiers,
+)
+
+
+class OutlierCheck:
+    def __init__(
+        self,
+        name: str,
+        columns_map: map,
+        accepted_std_div: int,
+        threshold: float,
+    ) -> None:
+        self.name = name
+        self.columns_map = columns_map
+        self.accepted_std_div = accepted_std_div
+        self.threshold = threshold
+
+
+class ChecksHandler:
+    def __init__(self, checks: List[OutlierCheck], conn_type: str):
+        self.checks = checks
+        self.conn_type = conn_type
+
+    def prepare_main_stats_postgres_sql(self, main_table: Table):
+        select_expressions = []
+        for check in self.checks:
+            for key in check.columns_map:
+                select_expressions.append(
+                    sql.SQL(
+                        "stddev({key}) as {check_name}_{key_name}_stddev, avg({key}) as {check_name}_{key_name}_avg"
+                    ).format(
+                        key=sql.Identifier(key),
+                        key_name=sql.SQL(key),
+                        check_name=sql.SQL(check.name),
+                    )
+                )
+
+        return sql.SQL(
+            "(SELECT {select_expressions} FROM {main_table}) as main_stats"
+        ).format(
+            select_expressions=sql.SQL(" ,").join(select_expressions),
+            main_table=sql.Identifier(main_table.table_name),
+        )
+
+    def prepare_main_stats_snowflake_sql(self, main_table: Table):
+        select_expressions = []
+        for check in self.checks:
+            for key in check.columns_map:
+                stats_query = "stddev({key}) as {check_name}_{key}_stddev, avg({key}) as {check_name}_{key}_avg"
+                replacements = {"{key}": key, "{check_name}": check.name}
+                for key, val in replacements.items():
+                    stats_query = stats_query.replace(key, val)
```

Comment:
Do we need to worry about the column names being SQL escapes as well?

Reply:
I'm not sure entirely, but they are used with 'AS' keyword anything besides an identifier should raise an syntax error.